coda/coda_mdstore/views.py

```diff
@@ -675,16 +675,19 @@ def externalIdentifierSearchJSON(request):
 
     identifiers = (External_Identifier.objects
                                       .select_related('belong_to_bag')
-                                      .filter(value=ark)
-                                      .distinct())
-    data = [
-        {
-            'name': exid.belong_to_bag.name,
-            'oxum': exid.belong_to_bag.oxum,
-            'bagging_date': exid.belong_to_bag.bagging_date.strftime('%Y-%m-%d')
-        }
-        for exid in identifiers
-    ]
+                                      .filter(value=ark))
+    unique_identifiers = []
+    data = []
+    for bagInfoObject in identifiers:
+        bag = bagInfoObject.belong_to_bag
+        if bag not in unique_identifiers:
+            unique_identifiers.append(bag)
+            data.append(
+                {
+                    'name': bag.name,
+                    'oxum': bag.oxum,
+                    'bagging_date': bag.bagging_date.strftime('%Y-%m-%d')
+                })
```

Comment:
Unlike in `externalIdentifierSearch`, we don't need that much data for each item returned. What do you think about changing the query, so `distinct` is useful? I am thinking something like:
```
    identifiers = (External_Identifier.objects
                                      .select_related('belong_to_bag')
                                      .filter(value=ark)
                                      .values_list('belong_to_bag__name',
                                                   'belong_to_bag__size',
                                                   'belong_to_bag__files',
                                                   'belong_to_bag__bagging_date')
                                      .distinct())

    data = [
        {
            'name': exid[0],
            'oxum': '%s.%s' % (exid[1], exid[2]),
            'bagging_date': exid[3].strftime('%Y-%m-%d')
        }
        for exid in identifiers
    ]
```

Reply:
oh..yeah..sure. I should have thought of it.